### PR TITLE
make broker interface more generic

### PR DESCRIPTION
**Pull request by bartosh**
Current implementation of --live-trading and --tws-uri command
line options is very IB-specific. We need something more generic
to be able to support different brokers.

Changed --live-trading and --tws-uri options to more generic
--broker and --broker-uri to be able to support more than one
broker interface.

Changed the way broker module is imported. From now on it's expected
that broker class will be placed into
   gens/brokers/<lowercase broker name>_broker.py
and broker module will be named <uppercase broker name>Broker,
e.g. zipline/gens/brokers/ib_broker.py and IBBroker.

This should hopefully allow us to add broker modules without
changing command line interface.

Signed-off-by: Ed Bartosh <bartosh@gmail.com>

### CODE REVIEW

File: zipline/utils/run_algo.py [hunk 1]
```diff
@@ -171,13 +167,12 @@ def choose_loader(column):
...
-    if live_trading:
+    if broker:
         start = pd.Timestamp.utcnow()
         end = start + pd.Timedelta('1', 'D')
-    TradingAlgorithmClass = (partial(LiveTradingAlgorithm,
-                                     live_trading=live_trading,
-                                     broker=broker)
-                             if live_trading else TradingAlgorithm)
+
+    TradingAlgorithmClass = (partial(LiveTradingAlgorithm, broker=broker)
```
Comment (tibkiss) [inline zipline/utils/run_algo.py]: The LiveTradingAlgorithm class needs to be adopted too as it is still trying to store live_trading parameter.

File: zipline/data/benchmarks.py [hunk 2]
```diff
@@ -12,50 +12,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import pandas as pd
```
Comment (tibkiss) [inline zipline/data/benchmarks.py]: Could you please create a separate PR for the GSPC - SPY related fixes?
It is an important fix, we shall not sneak in here :)

File: .travis.yml [hunk 3]
```diff
@@ -44,6 +44,7 @@ install:
   - pip install -r etc/requirements_dev.txt --cache-dir=$CACHE_DIR
   - pip install -r etc/requirements_blaze.txt --cache-dir=$CACHE_DIR  # this uses git requirements right now
   - pip install -r etc/requirements_talib.txt --cache-dir=$CACHE_DIR
+  - pip install -r etc/requirements-live-ib.txt --cache-dir=$CACHE_DIR
```
Comment (tibkiss) [inline .travis.yml]: This should be added to appveyor's settings too.

File: zipline/utils/run_algo.py (hunk 1 above)
Comment (bartosh) [inline zipline/utils/run_algo.py]: that's strange. I was able to run zipline with --broker ib. Will look at it. Thanks for pointing out!

**Comment by tibkiss** [comment]
I don't want to hold this PR back any further, but there is a merge conflict. Ed, could you please resolve the conflicts (or re-create the PR)? Thanks!

**Comment by tibkiss** [comment]
LGTM, thanks Ed!

**Pull request merged**